ui/raidboss/data/06-ew/dungeon/the_lunar_subterrane.ts

```diff
@@ -162,6 +172,111 @@ const triggerSet: TriggerSet<Data> = {
     },
+    {
+      // Round 1 is always a non-splitting line of orbs.
+      id: 'Lunar Subterrane Durante Forsaken Fount 1',
+      type: 'Ability',
+      netRegex: { id: '88BB', source: 'Durante', capture: false },
+      condition: (data) => data.fountsSeen === 0,
+      delaySeconds: 1, // This collides with Fount 2 if we don't delay. ???
+      infoText: (_data, _matches, output) => output.avoid!(),
+      run: (data) => data.fountsSeen += 1,
+      outputStrings: {
+        avoid: {
+          en: 'Away from orbs',
+        },
+      },
+    },
+    {
+      // Round 2 is always a splitting line. Center is always safe.
+      id: 'Lunar Subterrane Durante Forsaken Fount 2',
+      type: 'Ability',
+      netRegex: { id: '88BB', source: 'Durante', capture: false },
+      condition: (data) => data.fountsSeen === 1,
+      delaySeconds: 1,
+      response: Responses.goMiddle('info'),
+      run: (data) => data.fountsSeen += 1,
+    },
+    {
+      // On round three and subsequently, known  spawn locations for the Aetheric Charge orbs are:
+      // (0,-422)
+      // (0, -434.8)
+      // (0, -409.2)
+      // (3.65, -412.9)
+      // (-3.65, -412.95)
+      // (9.05, -412.8)
+      // (-9.05, -418.35)
+      // (9.15, -425.55)
+      // (9.15, -431.1)
+      // (-9.25, -431.1)
+      // (12.8, -422)
+      // (-12.8, -422)
+      // (Other spawn locations exist for rounds 1/2, but we can ignore those.)
+      id: 'Lunar Subterrane Durante Forsaken Fount 3 Collect',
+      type: 'AddedCombatant',
+      netRegex: { name: 'Aetheric Charge' },
+      condition: (data) => data.fountsSeen === 2,
+      run: (data, matches) => {
+        // Because the positions are relatively fixed, we don't need a reliable order for coordinates.
+        // Only the values and the count of those values really is important here.
+        data.fountX.push(Math.round(parseFloat(matches.x)));
+        data.fountY.push(Math.round(parseFloat(matches.y)) + 422); // Normalize the Y axis to 0
+      },
+    },
+    {
+      id: 'Lunar Subterrane Durante Forsaken Fount 3 Call',
+      type: 'AddedCombatant',
+      netRegex: { name: 'Aetheric Charge', capture: false },
+      condition: (data) => data.fountsSeen === 2,
+      delaySeconds: 0.5,
+      infoText: (data, _matches, output) => {
+        if (data.fountX.length < 5 || data.fountY.length < 5)
+          return;
+        // If there are five orbs on the field, three of them will, guaranteed,
+        // have the same X or Y value. Those three are in a straight line.
+        const hCount = data.fountY.filter((n) => n === 0).length;
+        const vCount = data.fountX.filter((n) => n === 0).length;
+
+        if (hCount === 3) {
+          // Horizontal lines are always east/west safe hammer patterns.
+          const xSum = data.fountX.reduce((a, b) => a + b, 0);
+
+          // Don't rely on the rounded sum to be precise, but the sign will be reliable.
+          if (xSum > 5)
```

Comment:
Can you give an example of what these >5, <5, >10, <10 patterns look like in terms of values? Summing makes sense as a way to differentiate between cases when you know the full set of examples, but it's a bit confusing to a reader. (Also, minorly and mostly curiously rather than something to fix, it makes me wonder if there's some simpler way to differentiate?)

Reply:
Sure, here's a raw list of configurations:

```
West Safe
(-12.80,-422.00), (0.00,-422.00), (12.80,-422.00)
(9.15,-431.10)
(3.65,-412.90)

North Safe
(0.00,-409.20), (0.00,-422.00), (0.00,-434.80)
(9.05,-412.80)
(-9.05,-418.35)

East Safe
(-12.80,-422.00),  (0.00,-422.00), (12.80,-422.00)
(-9.25,-431.05)
(-3.65,-412.95)

Hourglass
(0.00,-409.20), (0.00,-422.00), (0.00,-434.80)
(-9.05,-418.35)
(9.15,-425.55)
```

And Y-normalized:
```
West Safe
(-12.80,0), (0,0), (12.8,0)
(9.15,-9.1)
(3.65,9.1)

North Safe
(0,12.8), (0,0), (0,-12.8)
(9.05,9.2)
(-9.05,3.65)

East Safe
(-12.8,0),  (0,0), (12.8,0)
(-9.25,-9.05)
(-3.65,9.05)

Hourglass
(0,12.8), (0,0), (0,-12.8)
(-9.05,3.65)
(9.15,-3.55)
```

I'm not completely sure this list is exhaustive, but I'm confident the list of orb locations in the inline comment is.

Reply:
That makes sense! Can you add this as a comment somehow?

Reply:
Let's try this. I don't like how long it is, but the information is there. I also added some more in-line comments at the points where the output is actually returned.

Reply:
Thanks, I think that's helpful for future debugging, especially the "this probably does not exist" kind of comment.